### PR TITLE
allow configurable openshift-web-console image

### PR DESCRIPTION
**Pull request by philbrookes**
Deploy custom web-console image.

## Verification steps:
1) Install this version of the addon into minishift and enable it.
2) reconfigure minishift addon-env
```
minishift config set addon-env CONTAINER_REPO_USERNAME=${DOCKERHUB_USER},CONTAINER_REPO_PASSWORD="${DOCKERHUB_PASSWORD}",CONSOLE_IMAGE=aerogear/origin-web-console:integrations
```
3) recreate minishift
```
minishift stop
minishift delete
minishift start --openshift-version v3.9.0 --service-catalog
```
4)
Deploy fh-sync-server to a project and you should see the integrations section in the overview once fh-sync-server has successfully deployed.

### CODE REVIEW

File: mobile-core.addon [hunk 1]
```diff
@@ -26,7 +27,12 @@ echo Installing Ansible Service Broker: repo=#{CORE_REPO},branch=#{CORE_BRANCH},
 echo
 ssh curl -s -O -J -L https://raw.githubusercontent.com/#{CORE_REPO}/mobile-core/#{CORE_BRANCH}~/installer/roles/ansible-service-broker-setup/files/provision-ansible-service-broker.sh
 ssh chmod +x ./provision-ansible-service-broker.sh && ./provision-ansible-service-broker.sh '#{CONTAINER_REPO_USERNAME}' '#{CONTAINER_REPO_PASSWORD}' '#{CONTAINER_REPO_ORG}' true latest #{ip} nip.io ansible-service-broker
-ssh while [ $(oc get pods -n ansible-service-broker | grep "asb-" | grep -v "deploy" | grep "Running" | wc -l) -eq 0 ]; do sleep 10; done
+ssh while [ $(oc get pods -n ansible-service-broker | grep "asb-" | grep -v "deploy" | grep "Running" | wc -l) -eq 0 ]; do sleep 2; done
+
+echo Deploying web console image: #{CONSOLE_IMAGE}
+echo
+oc patch deployment webconsole -n openshift-web-console -p '{"spec": {"template": {"spec": {"containers": [{"name": "webconsole", "image": "#{CONSOLE_IMAGE}"}]}}}}'
+ssh while [ $(oc get pods -n openshift-web-console | grep "webconsole" | grep -v "deploy" | grep "Running" | wc -l) -eq 0 ]; do sleep 2; done
```
Comment (david-martin) [inline mobile-core.addon]: we could make the query more resilient by using labels to find the pod.

e.g. 
```
oc get pods -n openshift-web-console -l app=openshift-web-console --template "{{(index .items 0).status.phase}}"  | grep Running

Running
```

File: docs/create-custom-console-container.adoc [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+== Requirements
+Clone the link:https://github.com/openshift/origin-web-console-server[openshift-web-console-server] repo.
+
+== Steps
+First of all, in the origin-web-console repo, check out the code you wish to build an image for, and perform the link:https://github.com/openshift/origin-web-console#production-builds[required steps] to create an upto date `dist` directory.
+
+In the origin-web-console-server repo, run:
+```sh
+CONSOLE_REPO_PATH=path/to/origin-web-console make vendor-console
```
Comment (david-martin) [inline docs/create-custom-console-container.adoc:18]: This cmd gives an error about a missing dep

```
CONSOLE_REPO_PATH=~/work/origin-web-console make vendor-console
GIT_REF= CONSOLE_REPO_PATH=/home/dmartin/work/origin-web-console hack/vendor-console.sh
Making sure go-bindata binary has been built...
go-bindata/main.go:15:2: cannot find package "github.com/jteeuwen/go-bindata" in any of:
	/usr/lib/golang/src/github.com/jteeuwen/go-bindata (from $GOROOT)
	/home/dmartin/go/src/github.com/jteeuwen/go-bindata (from $GOPATH)
[ERROR] PID 11454: hack/vendor-console.sh:23: `go install ./...` exited with status 1.
[INFO] 		Stack Trace: 
[INFO] 		  1: hack/vendor-console.sh:23: `go install ./...`
[INFO]   Exiting with code 1.
make: *** [Makefile:137: vendor-console] Error 1
```

**Review by david-martin — APPROVED** [review]
Approved. Tried running a custom image locally, and building a custom image. Both are good.

Good to merge if we get a 2nd ack from@sedroche

File: docs/create-custom-console-container.adoc [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+make clean build
+OS_BUILD_ENV_PRESERVE=_output/local/bin hack/env make build-images
+docker tag openshift/origin-web-console:latest <YOUR_TAG_FOR_THIS_IMAGE>
+docker push <YOUR_TAG_FOR_THIS_IMAGE>
+```
+
+Now you are ready to start Minishift with the custom container
```
Comment (maleck13) [inline docs/create-custom-console-container.adoc:25]: Is there a link to this doc somewhere, I don't see it?

**Review by maleck13 — CHANGES_REQUESTED** [review]
I think we should very clearly call out in the README:

Setting up the mobile addon to view the latest UI changes with a set of instructions under that heading

**Pull request merged**